ivy/functional/backends/tensorflow/searching.py

```diff
@@ -111,7 +111,11 @@ def where(
     out: Optional[Union[tf.Tensor, tf.Variable]] = None,
 ) -> Union[tf.Tensor, tf.Variable]:
     x1, x2 = ivy.promote_types_of_inputs(x1, x2)
-    return tf.cast(tf.experimental.numpy.where(condition, x1, x2), x1.dtype)
+    dev = ivy.dev(x1, as_native=False)
+    ret = tf.cast(tf.experimental.numpy.where(condition, x1, x2), x1.dtype)
+    with tf.device(dev):
```

Comment:
Would we need to have this kind of logic for all functions? Probably cleaner to just use the device-shifting decorator in [this](https://github.com/unifyai/ivy/pull/16101) PR

Reply:
Yes, definitely a better way of handling it. Will wait for it to merge.